packages/launchpad/src/store/app.ts

```diff
@@ -0,0 +1,98 @@
+import { reactive, readonly, inject, App } from 'vue'
+
+interface State {
+  projectTitle: string
+  title: string
+  description: string
+  nextAction: () => void
+  backAction: () => void
+  alternativeAction?: () => void
+  steps: {
+      component?: boolean
+      setup?: boolean
+      dependencies?: boolean
+      configFile?: boolean
+
+      e2e?: boolean
+  }
+}
+
+function createInitialState (): State {
+  return {
+    projectTitle: 'design-system',
+    title: 'LaunchPad',
+    description: 'Scaffold Cypress Tests',
+    nextAction () {},
+    backAction () {},
+    steps: {},
+  }
+}
+
+const storeKey = Symbol('storeApp')
+
+export class StoreApp {
+  private readonly state: State;
+
+  install (app: App) {
+    app.provide(storeKey, this)
+  }
+
+  constructor (initialState: State) {
+    this.state = reactive(initialState)
+  }
+
+  getState () {
+    return readonly(this.state)
+  }
+
+  setMeta (meta: { title: string, description: string }) {
+    this.state.title = meta.title
+    this.state.description = meta.description
+  }
+
+  flagComponent (flag = true) {
```

Comment:
Do we need a default value? I think we should require a value to be passed explicitly.